agent/engine/dockerclient/dockerclientfactory.go

```diff
@@ -121,8 +135,54 @@ func findDockerVersions(endpoint string) map[DockerVersion]dockeriface.Client {
 		err = client.Ping()
 		if err != nil {
 			log.Infof("Failed to ping with Docker version %s: %v", version, err)
+			continue
 		}
 		clients[version] = client
 	}
 	return clients
 }
+
+func findDockerVersionsfromMinMaxVersions(endpoint string) (map[DockerVersion]dockeriface.Client, bool) {
```

Comment:
You're using the second parameter type (`bool`) to indicate the success/failure of this method, which means that returning an `error` instead is the better way to do that.

Reply:
I am not sure if it can be called a failure if min api version key is not found, nevertheless changed the return type to `error`